app/scripts/components/table/list-view.js

```diff
@@ -90,30 +93,25 @@ class List extends React.Component {
   }
 
   selectAll (e) {
-    const { rowId, list } = this.props;
-    const { data } = list;
-    const allSelected = this.state.selected.length === data.length;
-    if (!data.length) return;
-    else if (allSelected) {
-      this.setState({ selected: [] });
-    } else {
-      let selected;
-      if (typeof rowId === 'function') {
-        selected = data.map(rowId);
-      } else {
-        selected = data.map(d => d[this.props.rowId]);
-      }
+    const { rowId, list: { data } } = this.props;
+
+    if (!isEmpty(data)) {
+      const rowIdFn = isFunction(rowId) ? rowId : row => row[rowId];
+      const allSelected = this.state.selected.length === data.length;
```

Comment:
Definitely cleaner. I'm a bit confused why when `allSelected` is `true`, then we set `selected` to `[]`, but that seems to be what it was doing before

Reply:
That's because if everything is already selected and "select all" is clicked, we want to de-select all (i.e., select none).